gobblin-data-management/src/main/java/org/apache/gobblin/data/management/trash/Trash.java

```diff
@@ -189,11 +197,18 @@ public boolean moveToTrash(Path path) throws IOException {
     Path targetPathInTrash = PathUtils.mergePaths(this.trashLocation, fullyResolvedPath);
 
     if (!this.fs.exists(targetPathInTrash.getParent())) {
-      this.fs.mkdirs(targetPathInTrash.getParent());
+      if (this.simulate) {
+        LOG.info("Making a parent directory at " + targetPathInTrash.getParent() + " in trash.");
+      } else {
+        this.fs.mkdirs(targetPathInTrash.getParent());
+      }
     } else if (this.fs.exists(targetPathInTrash)) {
       targetPathInTrash = targetPathInTrash.suffix("_" + System.currentTimeMillis());
     }
-
+    if (this.simulate) {
+      LOG.info("Simulating moving " + fullyResolvedPath + " to " + targetPathInTrash + " in trash.");
+      return true;
+    }
```

Comment:
* nit , i prefer doing it like `else { return ...}` to emphasize on the need that only one blocks should run. in case we needed to add a line before L212, it may not by mistake be added to only one branch